### PR TITLE
Refactored tools and services so that they don't reference `PintaCore` directly

### PR DESCRIPTION
**Pull request by Lehonti**
This is a follow-up to #870 

A couple remarks, in order to help navigate these changes:

- As @cameronwhite mentioned, the constructor of the base class `ShapeTool` was calling a virtual method inside its constructor, which meant that the `services` field in each derived class was not initialized when the method was called, which resulted in an exception being thrown. This was (temporarily) solved by making the call to the method `Lazy`. See the constructor of `ShapeTool` and the fields above to see what this change entails.

- Additional to what we saw in that previous pull request, I removed `PintaCore` references from all core services. This involved changing some constructors so that they are passed _specific_ services that the instances depend on, but emphatically _not_ a `ServiceManager`, because we want to ensure that the services are initialized in the right order (see the static constructor of `PintaCore` to see how the constructors are called), and also some method signatures in some services, such that the services that could not be passed to the respective constructors would be passed as arguments instead. This should be fine in the sense that it should not cause any crashes / throw any exceptions, because all services exposed by `PintaCore` are initialized by the time its static constructor returns.

- There is still a lot of refactoring to be done, but my focus here was on removing `PintaCore` references from a certain key places while keeping the application functioning, otherwise the amount of changes would be extremely large. I plan to refactor further in future pull requests.

### CODE REVIEW

File: Pinta.Tools/Editable/EditEngines/ArrowedEditEngine.cs [hunk 1]
```diff
@@ -27,33 +27,32 @@
...
 using Pinta.Core;
 
 namespace Pinta.Tools;
 
 public abstract class ArrowedEditEngine : BaseEditEngine
 {
-	private Separator? arrow_sep;
-	private Label? arrow_label;
-	private CheckButton? show_arrow_one_box, show_arrow_two_box;
+	private Gtk.Separator? arrow_sep;
```
Comment (cameronwhite) [inline Pinta.Tools/Editable/EditEngines/ArrowedEditEngine.cs:36]: For future PRs, please avoid making extra changes like this which make it harder to review the real changes

File: tests/PintaBenchmarks/Mocks/MockPalette.cs [hunk 2]
```diff
@@ -8,6 +8,9 @@ internal sealed class MockPalette : IPaletteService
 	public Color PrimaryColor { get; set; } = new (0, 0, 0); // Black
 	public Color SecondaryColor { get; set; } = new (1, 1, 1); // White
 
+	public event EventHandler? PrimaryColorChanged;
```
Comment (cameronwhite) [inline tests/PintaBenchmarks/Mocks/MockPalette.cs:11]: There are some warnings here about unused events - can these be suppressed?

File: tests/Pinta.Effects.Tests/Mocks/MockPalette.cs [hunk 3]
```diff
@@ -8,6 +9,9 @@ internal sealed class MockPalette : IPaletteService
 	public Color PrimaryColor { get; set; } = new (0, 0, 0); // Black
 	public Color SecondaryColor { get; set; } = new (1, 1, 1); // White
 
+	public event EventHandler? PrimaryColorChanged;
```
Comment (cameronwhite) [inline tests/Pinta.Effects.Tests/Mocks/MockPalette.cs:12]: There are warnings here about unused events - can these be suppressed?

**Comment by Lehonti** [comment]
@cameronwhite I just suppressed the warnings :)

As for the changes that don't directly relate to what's being addressed in the PR: as you've seen, for quite some time now, along with the main changes in the PR in question I've been incorporating changes that are too small to have their own pull request, in order to save space (because Git stores entire files). If I don't do it that way, how could it be done?

**Comment by cameronwhite** [comment]
Thanks!
I don't know if there's a perfect answer, but it is more time-consuming to review large PRs  when there are other formatting / style changes mixed in. I'd limit that to small PRs, or style-only PRs (but those shouldn't be a high priority to work on IMO)

**Pull request merged**